core-bundle/src/Resources/contao/classes/BackendTemplate.php

```diff
@@ -138,6 +138,7 @@ protected function getLocaleString()
 				. '},'
 				. 'script_url:"' . TL_ASSETS_URL . '",'
 				. 'path:"' . \Environment::get('path') . '",'
+				. 'backend_index:"' . self::getContainer()->get('router')->generate('contao_backend') . '",'
```

Comment:
Wouldn’t it be better to directly use the `contao_backend_picker` route?

Reply:
absolutely!

Reply:
For solving the issue for the picker, yes. 

My idea was to provide a fix which would also enables other scripts to base on the base backend path. If it's not needed, we can use the route here.

Reply:
How about adding all routes here that start with `contao_backend_`?
@aschempp do you know an easy way to get all backend routes from the router?

Reply:
I also thought about a `Backend.route(name, params)` method which would make an background request to fetch the route. The benefit would be that params could be passed as well. But I thought it might be overhead for the actual issue.

Reply:
Sounds a bit overkill to me too…

Reply:
Could use the FOSJsRoutingBundle for that but imho it‘s really overkill.

Reply:
Adding the routes we currently need should be enough for now I think. Maybe as an object?

```js
"routes": {
	"contao_backend_picker": …,
}
```

Reply:
I changed this pull request implementing the solution @ausi proposed. I checked the core.js. So far as I see, there is only the `contao_backend_picker` route used. All other ajax requests depends on the current location href or an url parameter.